### PR TITLE
kazv: init at 0.5.0

### PR DESCRIPTION
**Pull request by fgaz**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ka/kazv/package.nix [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  cmake,
+  cmark,
+  cryptopp,
+  extra-cmake-modules,
+  immer,
+  kdePackages,
+  lager,
+  libkazv,
+  nlohmann_json,
+  olm,
```
Comment (gador) [inline pkgs/by-name/ka/kazv/package.nix:15]: Please be aware that as of https://github.com/NixOS/nixpkgs/pull/334638 `olm` is marked as insecure and both packages won't be build, unless you set `permittedInsecurePackages` or `export NIXPKGS_ALLOW_INSECURE=1`

Comment (fgaz) [reply pkgs/by-name/ka/kazv/package.nix:15]: I know, and that's fine by me. By the way, @tusooa (kazv dev) is spearheading the ecosystem-wide migration of c++ clients to vodozemac-bindings https://iron.lily-is.land/project/view/10/ :)

File: pkgs/by-name/li/libkazv/package.nix [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    libcpr
+    libhttpserver
+    libmicrohttpd
+    olm
+    nlohmann_json
+    zug
+  ];
+
+  cmakeFlags = [
+    # Catch2::Catch2WithMain target isn't found
```
Comment (SuperSandro2000) [inline pkgs/by-name/li/libkazv/package.nix:50]: Maybe because we have no catch2 in the dependencies?

Comment (tusooa) [reply pkgs/by-name/li/libkazv/package.nix:50]: I don't use nix but https://github.com/XYenon/nur-packages has a packaging of kazv & libkazv, you might want to refer to that.

File: pkgs/by-name/ka/kazv/package.nix (hunk 1 above)
Comment (tusooa) [reply pkgs/by-name/ka/kazv/package.nix:15]: the next release will be using vodozemac and no longer depend on olm.

File: pkgs/by-name/li/libkazv/package.nix (hunk 2 above)
Comment (tusooa) [reply pkgs/by-name/li/libkazv/package.nix:50]: https://github.com/XYenon/nur-packages/blob/master/pkgs/libkazv/default.nix

it's `catch2_3`, not `catch2`.

File: pkgs/by-name/ka/kazv/package.nix [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    kdePackages.kirigami
+    kdePackages.kirigami-addons
+    kdePackages.knotifications
+    lager
+    libkazv
+    nlohmann_json
+    olm
+    qt6.qtbase
+    qt6.qtmultimedia
+    qt6.qtwayland
```
Comment (tusooa) [inline pkgs/by-name/ka/kazv/package.nix:55]: qtimageformats is a recommended runtime dependency, without which it cannot display certain images like webp.

File: pkgs/by-name/li/libkazv/package.nix (hunk 2 above)
Comment (fgaz) [reply pkgs/by-name/li/libkazv/package.nix]: Thanks! fixed.

File: pkgs/by-name/ka/kazv/package.nix (hunk 3 above)
Comment (fgaz) [reply pkgs/by-name/ka/kazv/package.nix:56]: Added

File: pkgs/by-name/li/libkazv/package.nix [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  doCheck = true;
+
+  checkInputs = [ catch2_3 ];
+
+  meta = {
+    description = "Matrix client sdk built upon lager and the value-oriented design it enables";
+    homepage = "https://lily-is.land/kazv/libkazv";
+    license = lib.licenses.agpl3Plus;
+    maintainers = with lib.maintainers; [ fgaz ];
+    mainProgram = "libkazv";
```
Comment (fgaz) [inline pkgs/by-name/li/libkazv/package.nix:63]: oops, this was generated by nix-init and should be removed

**Comment by fgaz** [comment]
Ping

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**